docs/source/package_orderers.rst

```diff
@@ -0,0 +1,243 @@
+
+.. code-block:: python
+
+    package_orderers = [
+        {
+           "type": "per_family",
+           "orderers": [
+                {
+                    "packages": ["python"],
+                    "type": "version_split",
+                    "first_version": "2.7.16"
+                }
+            ]
+        }
+    ]
+
+
+soft_timestamp
+--------------
+
+This orderer takes in a given time ``T`` and returns packages released before ``T``, in descending order, followed by
+those released after.
+
+If ``rank`` is non-zero, version changes at that rank and above are allowed over the timestamp.
+
+A timestamp can be generated with python:
+
+.. code-block:: text
+
+   $ python -c "import datetime, time; print(int(time.mktime(datetime.date(2019, 9, 9).timetuple())))"
+   1568001600
+
+The following example will prefer package released before 2019-09-09.
+
+.. code-block:: python
+
+   package_orderers = [
+       {
+           "type": "soft_timestamp",
+           "timestamp": 1568001600,  # 2019-09-09
+           "rank": 3
+       }
+   ]
+
+The rank can be used to allow some versions released after the timestamp to still be considered.
+When using semantic versionnng, a value of 3 is the most common.
+This will let version with a different patch number to be accepted.
+
+Considering a package "foo" with the following versions:
+
+- "1.0.0" was released at 2019-09-07
+- "2.0.0" was released at 2019-09-08
+- "2.0.1" was released at 2019-09-10
+- "2.1.0" was released at 2019-09-11
+- "3.0.0" was released at 2019-09-12
+
+the following talbes shows the effect of rank:
+
+.. table::
+   :align: left
+
+   =========== ========== ==== =========
+   Example     Timestamp  Rank Result
+   =========== ========== ==== =========
+   rez-env foo 2019-09-09 0    foo-2.0.0
+   rez-env foo 2019-09-09 3    foo-2.0.1
+   rez-env foo 2019-09-09 2    foo-2.1.0
+   rez-env foo 2019-09-09 1    foo-3.0.0
+   =========== ========== ==== =========
+
+
+no_order
+--------
+
+An orderer that does not change the order - a no op.
+
+This orderer is useful in cases where you want to apply some default orderer
+to a set of packages, but may want to explicitly NOT reorder a particular
+package. You would use a :class:`rez.package_order.NullPackageOrder` in a :class:`rez.package_order.PerFamilyOrder` to do this.
+
+
+Custom orderers
+===============
+
+It is possible to create custom orderers using the API. This can be achieved
+by subclassing :class:`rez.package_order.PackageOrder` and implementing some mandatory
+methods. Once that's done, you need to register the orderer using :func:`rez.package_order.register_orderer`.
+
+.. note::
+
+   Implementing a custom orderer should only be done if absolutely necessary.
+   It could make your environment behave in very special ways and more importantly
+   in non expected ways from a user perspective. It can also make it harder to share
+   the set of affected packages to others.
+
+
+.. code-block:: python
```

Comment:
@JeanChristopheMorinPerso To clarify, the definition of the `MyOrderer` class here should NOT be in the `rezconfig.py` file right? This requires two different `.py` files doesn't it? If so we should probably clarify that.

Reply:
I should be in the rezconfig.